nixos/tests/zsh-history.nix

```diff
@@ -0,0 +1,38 @@
+import ./make-test-python.nix ({ pkgs, ...} : {
+  name = "zsh-history";
+  meta = with pkgs.stdenv.lib.maintainers; {
+    maintainers = [ kampka ];
+  };
+
+  nodes.default = { ... }: {
+    programs = {
+      zsh.enable = true;
+    };
+    environment.systemPackages = [ pkgs.zsh-history ];
+    programs.zsh.interactiveShellInit = ''
+      source ${pkgs.zsh-history.out}/share/zsh/init.zsh
+    '';
+  };
+
+  testScript = ''
+    start_all()
+    default.wait_for_unit("multi-user.target")
+    default.wait_until_succeeds("pgrep -f 'agetty.*tty1'")
+
+    # Login
+    default.wait_until_tty_matches(1, "login: ")
+    default.send_chars("root\n")
+    default.wait_until_tty_matches(1, "login: root")
+
+    # Launch a zsh shell. This should also load the zsh history code
+    default.send_chars("zsh\n")
+    default.wait_until_tty_matches(1, "root@default>")
```

Comment:
Can we set the default shell to zsh to remove this intermediary step?
Maybe it's https://nixos.org/nixos/options.html#users.defaultusershell, though I'm not entirely sure that will effect root.

Reply:
Seems to work, fixed.